src/LoggingConnector.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace LeProxy\LeProxy;
+
+use React\Socket\Connector;
+use React\Promise\Promise;
+use React\Stream\WritableStreamInterface;
+use React\Socket\ConnectorInterface;
+use React\Socket\ConnectionInterface;
+use Clue\React\HttpProxy\ProxyConnector as HttpClient;
+use Clue\React\Socks\Client as SocksClient;
+
+/** @internal */
+class LoggingConnector implements ConnectorInterface
+{
+    private $connector;
+    private $outputStream;
+    private $protocol = 'tcp';
+    private $auth = null;
+
+    public function __construct(ConnectorInterface $connector, WritableStreamInterface $outputStream)
+    {
+        if ($connector instanceof HttpClient) {
+            $this->protocol = 'http';
+        }
+
+        if ($connector instanceof SocksClient) {
+            $this->protocol = 'socks5';
+        }
```

Comment:
Not a big fan of these checks here, also doesn't look right. This check only ensures a connection from LeProxy to the next upstream proxy will be through "socks5". This is not to be confused with the protocol where this connection came *in* from.